22_pipeslab/pipeslab.txt

```diff
@@ -1,21 +1,23 @@
 For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
-  Answer: 
+  Answer: ls -l | tail -n +2 | cut -c4 | grep x | wc | cut -c6-7
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
-  Answer: 
+  Answer: ls -l | tail -n +2 | cut -c4,7,10 | grep x | wc | cut -c6-7
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
-  Answer: 
+  Answer: ls -l |tail -n +2 | grep $(date) | wc | cut -c6-7
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
+1: ls -l | tail -n +2 | cut -c1 | grep - | cut -c4 | grep x | wc | cut -c6-7
+2: ls -l | tail -n +2 | cut -c1 | grep - | cut -c4,7,10 | grep x | wc | cut -c6-7
 
-  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
-
+5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
+/sbin/ifconfig | grep 'inet ' | head -1
 6. Joe wanted to sort a text file in place so he tried to run the command "sort myfile.txt > myfile.txt". What happened to the file? Why?
-
+all the values that were in the file were deleted. Problem is that he tries to run sort on a file and not in the content. Should be something like "cat myfile.txt | sort"
```

Comment:
Not exactly. `sort file` works very well and prints its contents sorted. Also you can try `cat file.txt | sort > file.txt` and get the file deleted again.

The problem is the order bash performs output redirects. A `>` is first executed and truncates contents of the file. Then everything else is read and executed, but unfortunately file is already empty so cat / sort see no contents.

Sort has a switch for such a use case: `-o` Using it you can sort a file in place: `sort file.txt -o file.txt`